src/components/calcite-switch/calcite-switch.tsx

```diff
@@ -21,34 +22,107 @@ import { Scale, Theme } from "../interfaces";
   shadow: true
 })
 export class CalciteSwitch {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
   @Element() el: HTMLCalciteSwitchElement;
 
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
   /** True if the switch is disabled */
   @Prop({ reflect: true }) disabled?: boolean = false;
 
-  /** The name of the checkbox input */
-  @Prop({ reflect: true, mutable: true }) name?: string = "";
+  @Watch("disabled")
+  disabledWatcher(newDisabled: boolean): void {
+    this.inputEl.disabled = newDisabled;
+  }
+
+  /** The id attribute of the switch.  When omitted, a globally unique identifier is used. */
+  @Prop({ reflect: true, mutable: true }) guid: string;
```

Comment:
Shouldn't this be an internal state instead? We don't want users to set guid, correct?

Reply:
Yes, they would set the `id` attribute only.  Good catch.